django/thunderstore/api/cyberstorm/services/package_listing.py

```diff
@@ -0,0 +1,68 @@
+from django.core.exceptions import PermissionDenied
+from django.db import transaction
+from django.shortcuts import get_object_or_404
+
+from thunderstore.core.types import UserType
+from thunderstore.repository.models import Community, PackageListing
+from thunderstore.repository.views.package._utils import get_package_listing_or_404
+
+
+def get_package_listing(
+    namespace_id: str, package_name: str, community_id: str
+) -> PackageListing:
+    community = get_object_or_404(Community, identifier=community_id)
+    return get_package_listing_or_404(
+        namespace=namespace_id,
+        name=package_name,
+        community=community,
+    )
```

Comment:
This seems like it'd generate more queries than before (and loses some select_relateds which may or may not have been meaningful, I haven't verified).

Not sure how big of an impact this is in practice, but it could be a step back so making a note